server/pulp/server/managers/consumer/applicability.py

```diff
@@ -118,29 +121,48 @@ def regenerate_applicability_for_repos(repo_criteria):
         repo_criteria.fields = ['id']
         repo_ids = [r.repo_id for r in model.Repository.objects.find_by_criteria(repo_criteria)]
 
+        task_group_id = uuid4()
+
         for repo_id in repo_ids:
-            # Find all existing applicabilities for given repo_id. Setting batch size of 5 ensures
-            # the MongoDB cursor does not time out. See https://pulp.plan.io/issues/998#note-6 for
-            # more details.
-            existing_applicabilities = RepoProfileApplicability.get_collection().find(
-                {'repo_id': repo_id}).batch_size(5)
-            for existing_applicability in existing_applicabilities:
+            existing_applicability_ids = RepoProfileApplicability.get_collection().find(
+                {'repo_id': repo_id}, {'_id': 1})
+            for batch in paginate(existing_applicability_ids, 100):
```

Comment:
I recall that someone has measured this calculation at about 12 seconds per profile. I don't know how representative that is of production use, but it's probably in the neighborhood. 100 of those would take 20 minutes, which is a long time for a single task to run. I suggest going with more granularity, so workers have more opportunity to alternate with other kinds of work. For example, a batch size of 20 would take 4 minutes at that speed. I'd rather err on the side of being too granular, than to have users notice a long-running task and get annoyed by it.